### ISSUE TITLE
calling resumeTask from another task - libasync bug?

### ISSUE THREAD

**Issue by ricochet1k** [rejectedsoftware/vibe.d#1225]
This may be a bug in libasync, not sure who to blame. When vibe.d's configuration is left as the default, everything runs fine.

```
Building memutils 0.4.1 configuration "secure", build type debug.
Running dmd...
Enhanced memory security is enabled.
Building libasync 0.7.5 configuration "regular", build type debug.
Running dmd...
Enhanced memory security is enabled.
Using Linux EPOLL for events
Building vibe-d 0.7.24 configuration "libasync", build type debug.
Running dmd...
Enhanced memory security is enabled.
Using Linux EPOLL for events
Building blog.abcdefghij ~master configuration "application", build type debug.
Compiling using dmd...
Enhanced memory security is enabled.
Using Linux EPOLL for events
Compiling diet template 'index.dt'...
Linking...
Running ./blog.abcdefghij 
Listening for HTTP requests on ::1:8080
Listening for HTTP requests on 127.0.0.1:8080
Please open http://127.0.0.1:8080/ in your browser.
core.exception.AssertError@../../../../../../.dub/packages/vibe-d-0.7.24/source/vibe/core/core.d(1083): Calling resumeTask from another task.
----------------
./blog.abcdefghij(void vibe.core.core.VibeDriverCore.resumeTask(vibe.core.task.Task, Exception)+0xca) [0x810d5e]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncFileStream.handler()+0xeb) [0x8173af]
./blog.abcdefghij(shared(void function()) libasync.file.FileReadyHandler.opCall+0x74) [0x954038]
./blog.abcdefghij(shared(nothrow bool function(immutable(char)[], shared(ubyte[]), ulong, bool, bool)) libasync.file.AsyncFile.read+0x3f1) [0x952b29]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncFileStream.read(ubyte[])+0x1b1) [0x81680d]
./blog.abcdefghij(void vibe.core.stream.OutputStream.writeDefault(vibe.core.stream.InputStream, ulong)+0x124) [0x82d054]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncTCPConnection.write(vibe.core.stream.InputStream, ulong)+0x88) [0x81a8d4]
./blog.abcdefghij(void vibe.http.server.HTTPServerResponse.writeRawBody(vibe.core.stream.RandomAccessStream)+0x10c) [0x858c38]
./blog.abcdefghij(void vibe.http.fileserver.sendFile(scope vibe.http.server.HTTPServerRequest, scope vibe.http.server.HTTPServerResponse, vibe.inet.path.Path, vibe.http.fileserver.HTTPFileServerSettings)+0xbc7) [0x8c7417]
./blog.abcdefghij(void vibe.http.fileserver.serveStaticFile(vibe.inet.path.Path, vibe.http.fileserver.HTTPFileServerSettings).callback(scope vibe.http.server.HTTPServerRequest, scope vibe.http.server.HTTPServerResponse)+0x39) [0x8c6709]
./blog.abcdefghij(_D4vibe4http6router9URLRouter13handleRequestMFC4vibe4http6server17HTTPServerRequestC4vibe4http6server18HTTPServerResponseZ21__T9__lambda3TmTAAyaZ9__lambda3MFmMAAyaZv+0x1fb) [0x854fff]
./blog.abcdefghij(const(void function(immutable(char)[], scope void delegate(ulong, scope immutable(char)[][]))) vibe.http.router.MatchTree!(vibe.http.router.Route).MatchTree.doMatch+0x196) [0x7da196]
./blog.abcdefghij(void vibe.http.router.MatchTree!(vibe.http.router.Route).MatchTree.match(immutable(char)[], scope void delegate(ulong, scope immutable(char)[][]))+0x93) [0x7d99fb]
./blog.abcdefghij(void vibe.http.router.URLRouter.handleRequest(vibe.http.server.HTTPServerRequest, vibe.http.server.HTTPServerResponse)+0x196) [0x854d9a]
./blog.abcdefghij(bool vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool)+0x17d9) [0x8ca911]
./blog.abcdefghij(void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo)+0x1a3) [0x8c9033]
./blog.abcdefghij(void vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings).doListen(vibe.http.server.HTTPListenInfo, bool).__lambda3(vibe.core.net.TCPConnection)+0x36) [0x8c8a32]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncTCPConnection.onConnect()+0xa8) [0x81b7ec]
./blog.abcdefghij(void vibe.core.core.makeTaskFuncInfo!(void delegate()).makeTaskFuncInfo(ref void delegate()).callDelegate(vibe.core.core.TaskFuncInfo*)+0x72) [0x812036]
./blog.abcdefghij(void vibe.core.core.CoreTask.run()+0x129) [0x81001d]
./blog.abcdefghij(void core.thread.Fiber.run()+0x2a) [0x9c771e]
./blog.abcdefghij(fiber_entryPoint+0x61) [0x9c760d]
[(nil)]
Task terminated with unhandled exception: Calling resumeTask from another task.
core.exception.AssertError@../../../../../../.dub/packages/vibe-d-0.7.24/source/vibe/core/core.d(1083): Calling resumeTask from another task.
----------------
./blog.abcdefghij(void vibe.core.core.VibeDriverCore.resumeTask(vibe.core.task.Task, Exception)+0xca) [0x810d5e]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncFileStream.handler()+0xeb) [0x8173af]
./blog.abcdefghij(shared(void function()) libasync.file.FileReadyHandler.opCall+0x74) [0x954038]
./blog.abcdefghij(shared(nothrow bool function(immutable(char)[], shared(ubyte[]), ulong, bool, bool)) libasync.file.AsyncFile.read+0x3f1) [0x952b29]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncFileStream.read(ubyte[])+0x1b1) [0x81680d]
./blog.abcdefghij(void vibe.core.stream.OutputStream.writeDefault(vibe.core.stream.InputStream, ulong)+0x124) [0x82d054]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncTCPConnection.write(vibe.core.stream.InputStream, ulong)+0x88) [0x81a8d4]
./blog.abcdefghij(void vibe.http.server.HTTPServerResponse.writeRawBody(vibe.core.stream.RandomAccessStream)+0x10c) [0x858c38]
./blog.abcdefghij(void vibe.http.fileserver.sendFile(scope vibe.http.server.HTTPServerRequest, scope vibe.http.server.HTTPServerResponse, vibe.inet.path.Path, vibe.http.fileserver.HTTPFileServerSettings)+0xbc7) [0x8c7417]
./blog.abcdefghij(void vibe.http.fileserver.serveStaticFile(vibe.inet.path.Path, vibe.http.fileserver.HTTPFileServerSettings).callback(scope vibe.http.server.HTTPServerRequest, scope vibe.http.server.HTTPServerResponse)+0x39) [0x8c6709]
./blog.abcdefghij(_D4vibe4http6router9URLRouter13handleRequestMFC4vibe4http6server17HTTPServerRequestC4vibe4http6server18HTTPServerResponseZ21__T9__lambda3TmTAAyaZ9__lambda3MFmMAAyaZv+0x1fb) [0x854fff]
./blog.abcdefghij(const(void function(immutable(char)[], scope void delegate(ulong, scope immutable(char)[][]))) vibe.http.router.MatchTree!(vibe.http.router.Route).MatchTree.doMatch+0x196) [0x7da196]
./blog.abcdefghij(void vibe.http.router.MatchTree!(vibe.http.router.Route).MatchTree.match(immutable(char)[], scope void delegate(ulong, scope immutable(char)[][]))+0x93) [0x7d99fb]
./blog.abcdefghij(void vibe.http.router.URLRouter.handleRequest(vibe.http.server.HTTPServerRequest, vibe.http.server.HTTPServerResponse)+0x196) [0x854d9a]
./blog.abcdefghij(bool vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool)+0x17d9) [0x8ca911]
./blog.abcdefghij(void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo)+0x1a3) [0x8c9033]
./blog.abcdefghij(void vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings).doListen(vibe.http.server.HTTPListenInfo, bool).__lambda3(vibe.core.net.TCPConnection)+0x36) [0x8c8a32]
./blog.abcdefghij(void vibe.core.drivers.libasync.LibasyncTCPConnection.onConnect()+0xa8) [0x81b7ec]
./blog.abcdefghij(void vibe.core.core.makeTaskFuncInfo!(void delegate()).makeTaskFuncInfo(ref void delegate()).callDelegate(vibe.core.core.TaskFuncInfo*)+0x72) [0x812036]
./blog.abcdefghij(void vibe.core.core.CoreTask.run()+0x129) [0x81001d]
./blog.abcdefghij(void core.thread.Fiber.run()+0x2a) [0x9c771e]
./blog.abcdefghij(fiber_entryPoint+0x61) [0x9c760d]
[(nil)]
Error executing command run:
Program exited with code 1
```

**Comment by Positronic** [comment]
I basically had the same error but after hours chasing down what I thought was some error on my end (new to vibe.d) and/or my system (Win 10, x84_64 using VS2015 CRT) I eventually ended up here.  After applying etcimon's commit 88bc8d2 to my v0.7.25 libasync.d it works!

Now, I may be a GitHub noob here, but looking at the version tree it looks like back on Aug 16 the commit was added to it's sole branch, labeled "patch-19" and then .... nothing?  Am I reading that wrong?  If not, is there a reason this wasn't merged into 0.7.25, or when can we expect it to be merged, next release?
